### DISCUSSION TITLE
[@types/node] Promise is missing withResolvers in node.js 22

### DISCUSSION THREAD

**Discussion by ZatsuneSakurako** [DefinitelyTyped/DefinitelyTyped#71526] [Issues with a @types package] [answered]
In Node.js 22, the support of `Promise.withResolvers()` (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/withResolvers) has been added but the corresponding definitions does not exist in @types/node 22.X

[a comment by an automated account is not shown]

**Comment by Renegade334** [comment, marked as answer]
ES features are included with TypeScript's own libraries rather than @types/node. If you want to use this feature, include [es2024.promise](https://github.com/microsoft/TypeScript/blob/main/src/lib/es2024.promise.d.ts) in your tsconfig libs.

**Reply by ZatsuneSakurako** [reply]
Oh yes, it works, thanks
